2.4 Data Flow.md

```diff
@@ -67,11 +67,12 @@ Then, the following objective checks are done:
 After the objective checks, the following subjective checks are done:
 
 5. Inputs' branches rejection check: it checks if all the consumed inputs belong to a non-rejected branch. Notice that this is not an objective check, so the node is susceptible (even if with a small probability) to have its opinion about rejected branches changed by a reorganization. For that reason, if the message does not pass this check, the message is booked into the Tangle and ledger state (even though the balances are not altered by this message, since it will be booked to a rejected branch). This is what we call "lazy booking", which is done to avoid huge re-calculations in case of a reorganization of the ledger. If it passes the check, it goes to the next step. 
-10. Double spend check: it checks if any of the inputs is conflicting with a transaction that was already confirmed. As in the last step, this check is not objective and, thus, if the message does not pass this check, it is lazy booked into the Tangle and ledger state, into an invalid branch. If it passes the check, it goes to the next step. 
+6. Double spend check: it checks if any of the inputs is conflicting with a transaction that was already confirmed. As in the last step, this check is not objective and, thus, if the message does not pass this check, it is lazy booked into the Tangle and ledger state, into an invalid branch. If it passes the check, it goes to the next step. 
+7. Weak Past cone check: it checks if the inputs of the transaction were created by transactions that are either directly referenced by the transaction or already confirmed. Note the transaction may not be confirmed for all nodes at the time of the check. If the transaction does not fulfil these requirements it is temporarily queued until the transaction containing the inputs is confirmed. Once it passes the check, it goes to the next step. 
```

Comment:
@darcy-camargo i have modified also the data flow point 7 above now. Since the weak past cone check is a requirement and because we removed the objective past cone check

Reply:
Lets go through with this, but this spec will be updated later as we remove the weak approvals.

Reply:
the weak past cone check has actually nothing to do with weak approvals. i have renamed it to be more clear.